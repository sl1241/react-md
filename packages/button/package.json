{
  "name": "@react-md/button",
  "version": "5.1.3",
  "description": "This package is used to create buttons with the material design spec. There is built-in support to be able to render accessible clickable divs that look like buttons, or even apply button styles to any element.",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./types/index.d.ts",
  "sideEffects": [
    "dist/**/*"
  ],
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md.git",
    "directory": "packages/button"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "homepage": "https://react-md.dev/packages/button/demos",
  "keywords": [
    "accessibility",
    "a11y",
    "react-md",
    "material design",
    "react",
    "button",
    "component",
    "components"
  ],
  "license": "MIT",
  "dependencies": {
    "@react-md/elevation": "workspace:../elevation",
    "@react-md/icon": "workspace:../icon",
    "@react-md/states": "workspace:../states",
    "@react-md/theme": "workspace:../theme",
    "@react-md/transition": "workspace:../transition",
    "@react-md/typography": "workspace:../typography",
    "@react-md/utils": "workspace:../utils",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@types/react": "^18.0.29",
    "csstype": "^3.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "peerDependencies": {
    "react": ">= 16.14",
    "react-dom": ">= 16.14"
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": false
    },
    "react-dom": {
      "optional": false
    }
  },
  "publishConfig": {
    "access": "public"
  }
}
