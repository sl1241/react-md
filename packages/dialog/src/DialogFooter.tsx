


import { bem } from "@react-md/utils"

/**
 * An optional alignment for the content within the footer. Since the majority
 * of dialog footers are used to contain action buttons, the default alignment
 * is near the end.
 *
 * @remarks \@since 3.1.0
 */
export type DialogFooterAlignment =
  | "none"
  | "start"
  | "end"
  | "between"
  | "stacked-start"
  | "stacked-end"

export interface DialogFooterProps extends HTMLAttributes<HTMLDivElement> {
  /** {@inheritDoc DialogFooterAlignment} */
  align?: FunctionMaybe<Nullable<DialogFooterAlignment>>
}

const block = bem("rmd-dialog")

export const DialogFooter = ({ children, className, align = "end", ref, ...props }: DialogFooterProps) => {
  return <footer
    {...props}
    //@ts-ignore
    ref={ref}
    className={[
      block("footer", {
        flex: align !== "none",
        "flex-v": align === "stacked-start" || align === "stacked-end",
        start: align === "start" || align === "stacked-start",
        between: align === "between",
        end: align === "end" || align === "stacked-end",
      }),
      className
    ]}
  >
    {children}
  </footer>
}

