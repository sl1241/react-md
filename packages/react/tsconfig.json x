{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
