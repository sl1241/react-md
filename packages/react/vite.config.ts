import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
