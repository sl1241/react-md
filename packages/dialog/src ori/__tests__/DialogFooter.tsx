import type { ReactElement } from "react";
import { Button } from "@react-md/button";
import { render } from "@testing-library/react";

import type { DialogFooterAlignment, DialogFooterProps } from "../DialogFooter";
import { DialogFooter } from "../DialogFooter";

const aligns: readonly DialogFooterAlignment[] = [
  "none",
  "start",
  "end",
  "between",
  "stacked-start",
  "stacked-end",
];

function Test({ align }: DialogFooterProps): ReactElement {
  return (
    <DialogFooter align={align}>
      <Button>First Button</Button>
      <Button>Second Button</Button>
    </DialogFooter>
  );
}

describe("DialogFooter", () => {
  it("should render correctly", () => {
    const { container, rerender } = render(<Test />);
    expect(container).toMatchSnapshot();

    aligns.forEach((align) => {
      rerender(<Test align={align} />);
      expect(container).toMatchSnapshot();
    });
  });
});
