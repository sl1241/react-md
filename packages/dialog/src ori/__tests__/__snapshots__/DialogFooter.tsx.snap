// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogFooter should render correctly 1`] = `
<div>
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--end"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;

exports[`DialogFooter should render correctly 2`] = `
<div>
  <footer
    class="rmd-dialog__footer"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;

exports[`DialogFooter should render correctly 3`] = `
<div>
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--start"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;

exports[`DialogFooter should render correctly 4`] = `
<div>
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--end"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;

exports[`DialogFooter should render correctly 5`] = `
<div>
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--between"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;

exports[`DialogFooter should render correctly 6`] = `
<div>
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--flex-v rmd-dialog__footer--start"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;

exports[`DialogFooter should render correctly 7`] = `
<div>
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--flex-v rmd-dialog__footer--end"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      First Button
      <span
        class="rmd-ripple-container"
      />
    </button>
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Second Button
      <span
        class="rmd-ripple-container"
      />
    </button>
  </footer>
</div>
`;
